app/routes/goals.py

```diff
@@ -0,0 +1,161 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from datetime import datetime
+from flask import Blueprint, jsonify, make_response, request, abort
+
+goal_bp =  Blueprint('goals', __name__, url_prefix= "/goals")
+
+
+# POST create a new  goal
+@goal_bp.route('', methods = ['POST'])
+def create_goal():
+    request_body = request.get_json()
+
+    if "title" not in request_body:
+        response_body = {"details": "Invalid data. Title is required"} 
+        return make_response(response_body, 400)
+    try:
+        new_goal = Goal(title=request_body["title"])
+        # add new entity and commit it 
+        db.session.add(new_goal)
+        db.session.commit()
+
+        response_body = {
+            "goal": new_goal.to_dict()
+        }
+        return make_response(jsonify(response_body),201)  
+    except:
+        abort(422)
```

Comment:
Providing error messages instead of just error codes helps with debugging, also 400 or 500 are probably more appropriate here:

```suggestion
        abort(make_response({"error": "could not create goal"}, 400))
```

(422 is a code specific to WebDAV.)